asynq/async_task.py

```diff
@@ -182,8 +140,10 @@ def _continue(self):
             except BaseException as error:
                 self._accept_error(error)
             finally:
-                if self._dependencies or self.is_computed():
-                    self._after_continue()
+                if self.is_computed():
+                    self._dependencies = []
+                    return
```

Comment:
Returning in a finally block is risky, because exceptions finally get removed if you return. In this case, no exceptions can happen because we catch all exceptions right above, but that also means that this doesn't need to be a finally block. Maybe we should move this part out of a finally block. (The same problem was present before your change, so if you prefer I can submit a separate PR to fix it after we merge this one.)